### DISCUSSION TITLE
Some tips on handling line endings on Windows

### DISCUSSION THREAD

**Discussion by malkia** [bazelbuild/bazel-central-registry#1828] [General]
Not really much to say, just something to be aware when on Windows.

By default `git` tends to convert text files from LF ("unix") to CR/LF ("dos"). This changes the the integrity checksums.

What helps, is bit obvious - disable CR/LF conversion for this repo, or even globally!

Another way, is use the `dos2unix` (and there is `unix2dos`) from  `C:\Program Files\Git\usr\bin\` (not in your PATH by default), to convert locally

Probably sanest thing is to have the repo checkedout "unix" way.

**Comment by malkia** [comment]
Or maybe some kind of "linter" can be done for the repo - where it checks that the stuff is already in the proper "unix" line ending?

**Reply by fmeum** [reply]
We could add a `.gitattributes` file that disables CR/LF conversion for the entire repo. Would you be interested in submitting a PR?

**Reply by malkia** [reply]
Ah, just learened about this yesterday - not sure if this is the right thing - but here it is - https://github.com/bazelbuild/bazel-central-registry/pull/1835 - Thanks!!!
